vulkan_helpers/vulkan_application.cpp

```diff
@@ -450,6 +450,12 @@ VkDevice VulkanApplication::CreateDevice(
                      use_sparse_binding);
 }
 
+void VulkanApplication::InitializationComplete() {
+  if (entry_data_->write_pipeline_cache()) {
+    WritePipelineCache(&device_, &pipeline_cache_, entry_data_->write_pipeline_cache());
```

Comment:
The handles are actually wrappers around the Vulkan handles. 
```
using VkPipelineCache = VkSubObject<PipelineCacheTraits, DeviceTraits>;
```
This allows us to automatically clean them up, (as well as hang function pointers off of dispatchable objects). 

Since these are "effectively" unique_ptr style objects, we cannot pass by value, and google style guide disallows us to pass by non-const ref (would have been my preference for clarity).
So we pass by pointer.